staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -214,6 +217,10 @@ func ModifyConfig(configAccess ConfigAccess, newConfig clientcmdapi.Config, rela
 				}
 			}
 
+			if lockErrors[destinationFile] != nil {
```

Comment:
move this above the getConfigFromFile call... we want to make sure no one else is writing before we read in the file to get the current content we're going to rewrite

Reply:
Sure you're totally right: Fixed in ecf3b83e88ef4c4791889f922514cf459d3b3058